task_processing/plugins/persistence/dynamodb_persistence.py

```diff
@@ -56,26 +59,21 @@ def _event_to_item(self, e):
                         resp[k] = []
                         for i in v:
                             resp[k].append(self._event_to_item(i))
-                elif type(v) is datetime.datetime:
-                    resp[k] = {
-                        'N': str(int(v.timestamp()))
-                    }
-                elif type(v) is bool:
-                    resp[k] = {
-                        'BOOL': v
-                    }
             return resp
         elif type(raw) is str:
             return {
                 'S': raw
             }
         elif type(raw) is int:
             return {
-                'I': str(raw)
+                'N': str(raw)
             }
         else:
             print("Missed converting key %s type %s" % (raw, type(raw)))
 
+    def item_to_event(self, obj):
+        return self._replace_decimals(obj)
```

Comment:
i'm guessing here we need this because type check fails on precord for int/float fields? it's probably "better" to use precord field factories instead of conflating ints and floats here.

    field(factory=xxx) <- will call xxx() on whatever value supplied to intializer